### PR TITLE
Update and enhance README

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Since we have now NFD as part of the official documentation, is time to update and enhance the README to be more dev focused, and all the user focus documentation will be at the openshift official docs page
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,60 @@
...
+daemonset.apps/nfd-worker   3         3         3       3            3           <none>          5s
+NAME                         READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/nfd-master   1/1     1            1           17s
 ```
-$ make test-e2e
+
+Check that NFD feature labels have been created
+
+```bash
+$ oc get no -o json | jq .items[].metadata.labels
```
Comment (kpouget) [inline README.md:50]: `oc get nodes`?

Comment (ArangoGutierrez) [reply README.md:50]: I really like short names in kube, looks like you don't  jejeje , is ok

File: README.md [hunk 2]
```diff
@@ -1,56 +1,60 @@
...
+> Prerequisite: a running OpenShift cluster 4.6+ 
 
-Update the Dockerfile with the correct golang build image and the base image the operator will run with.
-
-## Manual deploy of the operator
-Checkout the sources
+Get the source code
 
+```bash
+git clone -b {{ site.release }} https://github.com/openshift/cluster-nfd-operator
```
Comment (kpouget) [inline README.md:17]: is this supposed to be transformed by a template engine? 
it seems it doesn't work for the README: https://github.com/ArangoGutierrez/cluster-nfd-operator/tree/devel/readme49

Comment (ArangoGutierrez) [reply README.md:17]: good point, will fix

File: README.md (hunk 1 above)
Comment (kpouget) [reply README.md:50]: `no` ;-)
but that's true I don't use it much, except `ds` I guess

**Comment by kpouget** [comment]
/lgtm

thanks @ArangoGutierrez :)

**Pull request merged**